account_billing/models/account_billing.py

```diff
@@ -0,0 +1,194 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Ecosoft Co., Ltd (http://ecosoft.co.th/)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+import time
+from odoo import models, fields, api, _
+from odoo.exceptions import ValidationError
+
+MAP_INVOICE_TYPE_PARTNER_TYPE = {
+    'out_invoice': 'customer',
+    'in_invoice': 'supplier',
+}
+
+
+class AccountBilling(models.Model):
+    _name = 'account.billing'
+    _description = 'Account Billing'
+    _inherit = ['mail.thread']
+    _order = 'date desc, id desc'
+
+    number = fields.Char(
+        readonly=True,
+        copy=False,
+        help='Number of account.billing',
+    )
+    partner_id = fields.Many2one(
+        'res.partner',
+        string='Partner',
+        required=True,
+        default=lambda self: self._get_partner_id(),
+        help='Partner Information',
+        track_visibility='always'
+    )
+    date = fields.Date(
+        string='Billing Date',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        default=lambda self: time.strftime('%Y-%m-%d'),
+        help='Effective date for accounting entries',
+        track_visibility='always'
+    )
+    due_date = fields.Date(
+        string='Due Date',
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+        default=lambda self: fields.Date.context_today(self),
+        required=True,
+        track_visibility='always'
+    )
+    invoice_ids = fields.One2many(
+        'account.invoice',
```

Comment:
Make it explicit. I think it is become v12 coding standard now.
```suggestion
        'account.invoice',
        comodel_name='account.invoice',
        inverse_name='billing_id',
```